index.js

```diff
@@ -38,12 +39,21 @@ Cloneable.prototype.clone = function () {
   }
 
   this._clonesCount++
-  return pump(this, new Clone(this))
+  return this.pipe(new Clone(this))
+}
+
+function forwardDestroy (src, dest) {
+  src.on('error', destroy)
+  src.on('close', destroy)
+
+  function destroy (err) {
+    dest.destroy(err)
+  }
```

Comment:
This works, but feels hacky:

```js
eos(src, { writable: false }, function (err) {
  if (/premature close/.test(err)) dest.destroy()
  else if (err) dest.destroy(err)
})
```

This way, we interpret a premature close - meaning a `close` event was emitted before `end` - as the source having been destroyed.